nuitka/utils/CStrings.py

```diff
@@ -28,34 +28,31 @@ def _encodePythonStringToC(value):
     """
     assert type(value) is bytes, type(value)
 
-    result = ""
+    special_chars = {92: r"\134", 9: r"\011", 13: r"\015", 10: r"\012", 63: r"\077", 34: r"\042"}
+    result = ['"']
     octal = False
-
     for c in value:
-        if str is bytes:
-            cv = ord(c)
-        else:
-            cv = c
-
-        if c in b'\\\t\r\n"?':
-            result += r"\%03o" % cv
+        cv = c
 
+        if c in special_chars:
+            result.append(special_chars[c])
             octal = True
+
         elif 32 <= cv <= 127:
             if octal and c in b"0123456789":
-                result += '" "'
-
-            result += chr(cv)
+                result.append(r'" "')
 
+            result.append(chr(cv))
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (bug_risk):** Appending a space-quoted string directly might introduce bugs.

Consider the implications of directly appending `" "` without checking the context first.